### PR TITLE
add config for whether encoding number of primary keys

### PR DESCRIPTION
**Pull request by zhoney**
Change-Id: I07e113b7f29e0d38e5d362663046c755ea8dc751

### CODE REVIEW

File: hugegraph-core/src/main/java/com/baidu/hugegraph/structure/HugeVertex.java [hunk 1]
```diff
@@ -200,12 +202,14 @@ public void correctVertexLabel(VertexLabel correctLabel) {
                         IdStrategy.PRIMARY_KEY);
 
         List<Object> propValues = new ArrayList<>(primaryKeys.size());
+        boolean encodeNumber = ((HugeConfig) this.graph().configuration()).get(
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/structure/HugeVertex.java:205]: don't call configuration() if enable auth

Comment (javeme) [reply hugegraph-core/src/main/java/com/baidu/hugegraph/structure/HugeVertex.java:205]: add method graph.option(CoreOptions.VERTEX_ENCODE_PK_NUMBER) and set whitelist

**Comment by javeme** [comment]
please set related issue id

File: hugegraph-api/src/main/java/com/baidu/hugegraph/auth/HugeGraphAuthProxy.java [hunk 2]
```diff
@@ -547,6 +548,11 @@ public long now() {
         return this.hugegraph.now();
     }
 
+    @Override
+    public <V> V option(ConfigOption<V> option) {
+        return this.hugegraph.option(option);
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/auth/HugeGraphAuthProxy.java:553]: this.verifyStatusPermission();

File: hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/SchemaTransaction.java [hunk 3]
```diff
@@ -270,6 +270,11 @@ public boolean existsSchemaId(HugeType type, Id id) {
         return this.getSchema(type, id) != null;
     }
 
+    private boolean syncDelete() {
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/backend/tx/SchemaTransaction.java:273]: let asyncRun get it by option()

File: hugegraph-core/src/main/java/com/baidu/hugegraph/schema/builder/AbstractBuilder.java [hunk 4]
```diff
@@ -95,7 +96,7 @@ protected void checkSchemaIdIfRestoringMode(HugeType type, Id id) {
     }
 
     protected long taskWaitTimeout() {
-        return this.transaction.taskWaitTimeout();
+        return this.graph.option(CoreOptions.TASK_WAIT_TIMEOUT);
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/schema/builder/AbstractBuilder.java:99]: delete taskWaitTimeout() method

File: hugegraph-core/src/main/java/com/baidu/hugegraph/structure/HugeFeatures.java [hunk 5]
```diff
@@ -279,8 +277,8 @@ public boolean supportsDefaultLabel() {
         }
 
         public String defaultLabel() {
-            Configuration conf = HugeFeatures.this.graph.configuration();
-            return ((HugeConfig) conf).get(CoreOptions.VERTEX_DEFAULT_LABEL);
+            return HugeFeatures.this.graph
+                                    .option(CoreOptions.VERTEX_DEFAULT_LABEL);
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/structure/HugeFeatures.java:281]: align with ".this"

File: hugegraph-core/src/main/java/com/baidu/hugegraph/structure/HugeVertex.java [hunk 6]
```diff
@@ -200,12 +201,14 @@ public void correctVertexLabel(VertexLabel correctLabel) {
                         IdStrategy.PRIMARY_KEY);
 
         List<Object> propValues = new ArrayList<>(primaryKeys.size());
+        boolean encodeNumber = this.graph()
+                                   .option(CoreOptions.VERTEX_ENCODE_PK_NUMBER);
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/structure/HugeVertex.java:205]: move to line 203

File: hugegraph-core/src/main/java/com/baidu/hugegraph/task/HugeTask.java [hunk 7]
```diff
@@ -635,10 +637,13 @@ private void checkPropertySize(String property, String propertyName) {
 
     private void checkPropertySize(int propertyLength, String propertyName) {
         int propertyLimit = BytesBuffer.STRING_LEN_MAX;
+        HugeGraph graph = this.scheduler().graph();
         if (propertyName.equals(P.INPUT)) {
-            propertyLimit = this.scheduler().taskInputSizeLimit();
+            propertyLimit = graph.option(CoreOptions.TASK_INPUT_SIZE_LIMIT)
+                                 .intValue();
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/task/HugeTask.java:643]: change propertyLimit to long

**Review by javeme — CHANGES_REQUESTED** [review]

**Review by javeme — APPROVED** [review]

**Review by Linary — APPROVED** [review]

**Pull request merged**